### ISSUE TITLE
Changing Config doesn't seem to do anything.

### ISSUE THREAD

**Issue by iv8512** [mschae23/grind-enchantments#55]
I read one of the older issues that says that you need to edit both the client config and the server one but still it doesn't seem to do anything. Even when going singleplayer and changing the config, the number of levels required always stays the same. 
I even tried to change the function type. I tested with the same item every time but the amount always stayed at 17 levels.

An easier way to make everything cheaper would be great, or at least some documentation on how to do so.

Here is my current config.
```
{
"allow_removing_curses": false,
"dedicated_server_options": {
"alternative_cost_display_enabled": false
},
"client_options": {
"show_enchantment_cost": true
},
"disenchant_to_book": {
"enabled": true,
"consume_enchanted_item": false,
"cost_function": {
"function": {
"type": "grindenchantments:count_levels"
},
"factor": 1.0,
"offset": 0.0,
"type": "grindenchantments:transform"
}
},
"move_enchantments": {
"enabled": true,
"cost_function": {
"function": {
"function": {
"normal_factor": 1.0,
"treasure_factor": 2.0,
"type": "grindenchantments:count_levels"
},
"type": "grindenchantments:first_enchantment"
},
"factor": 0.5,
"offset": 0.5,
"type": "grindenchantments:transform"
}
},
"version": 2
}
```

**Comment by chonkyman** [comment]
I agree, by default everything is a bit too expensive. I had the same problem and came to the GitHub to see if there was a way to tweak this.

**Comment by mschae23** [comment]
Do you see any related errors in the log (`.minecraft/logs/latest.log`)? If any error occurs while reading the config file, the mod is just going to use the default.

Also, what Minecraft version are you on? This looks like an older config.

> An easier way to make everything cheaper would be great, or at least some documentation on how to do so.

Yeah, this is severely lacking in that regard. I plan on writing some documentation at some point, but until then, there's not really much other than [this comment on another issue](https://github.com/mschae23/grind-enchantments/issues/46#issuecomment-1646937189).

**Comment by MeeniMc** [comment]
you need to add 

```
   "disenchant_to_book": {
   ...
+     "normal_factor": 1.0,
+     "treasure_factor": 2.0, 
      "type": "grindenchantments:count_levels"
```

**Comment by Gomezb3** [comment]
> you need to add
> 
> ```
>    "disenchant_to_book": {
>    ...
> +     "normal_factor": 1.0,
> +     "treasure_factor": 2.0, 
>       "type": "grindenchantments:count_levels"
> ```

I've been having this same issue, no matter what values I put or when I changed types for disenchanting to book, the values would not change. I am using version 4.0.0+1.21.1. By adding the lines normal_factor and treasure_factor, everything seems to work now and the cost is much lower. 

To check everything, I deleted the config on my end just to double check I didn't accidentally delete it, and the mod does not add these lines upon recreating the config file. I did check the logs when loading into my server and it received the server config, however what it received wasn't what it should be. I had previously changed it from count_min_power to count_levels, yet in the log it still said CountMinPower. 

Prior to receiving the server config in the logs, it did give out this one error:
`[00:59:01] [Render thread/ERROR]: [Grind Enchantments] Error decoding config: No key treasure_factor in MapLike[{"type":"grindenchantments:count_levels"}]; No key normal_factor in MapLike[{"type":"grindenchantments:count_levels"}]`

Based on my limited understanding, this does seem to confirm that the normal_factor and treasure_factor lines missing under disenchant_to_book are causing the issue with the config not changing properly and upon adding them manually, everything seems to work as it should.

**Comment by mschae23** [comment]
Closing this because it's been stale for a while. I have documentation on the config format now (<https://docs.mschae23.de/Grind_enchantments/Config>), but if you have any further questions, feel free to reopen this or open a new issue!
